examples/step-58/doc/intro.dox

```diff
@@ -0,0 +1,840 @@
+understand this equation further, take the time derivative of one of
+the equations, say
+@f{align*}{
+  \frac{\partial^2 w}{\partial t^2}
+  - \frac 12 \Delta \frac{\partial v}{\partial t}
+  &= 0,
+@f}
+(where we have assumed that, at least in some formal sense, we can
+commute the spatial and temporal derivatives), and then insert the
+other equation into it:
+@f{align*}{
+  \frac{\partial^2 w}{\partial t^2}
+  + \frac 14 \Delta^2 w
+  &= 0.
+@f}
+This equation is hyperbolic and similar in character to the wave
+equation. (This will also be obvious if you look at the video
+in the "Results" section of this program.) Furthermore, we could
+have arrived at the same equation for $v$ as well.
+Consequently, a better assumption for the NLSE is to think of
+it as a hyperbolic, wave-propagation equation than as a diffusion
+equation such as the heat equation. (You may wonder whether it is
+correct that the operator $\Delta^2$ appears with a positive sign
+whereas in the wave equation, $\Delta$ has a negative sign. This is
+indeed correct: After multiplying by a test function and integrating
+by parts, we want to come out with a positive (semi-)definite
+form. So, from $-\Delta u$ we obtain $+(\nabla v,\nabla u)$. Likewise,
+after integrating by parts twice, we obtain from $+\Delta^2 u$ the
+form $+(\Delta v,\Delta u)$. In both cases do we get the desired positive
+sign.)
+
+The real NLSE, of course, also has the terms $V\psi$ and
+$\kappa|\psi|^2\psi$. However, these are of lower order in the spatial
+derivatives, and while they are obviously important, they do not
+change the character of the equation.
+
+In any case, the purpose of this discussion is to figure out
+what time stepping scheme might be appropriate for the equation. The
+conclusions is that, as a hyperbolic-kind of equation, we need to
+choose a time step that satisfies a CFL-type condition. If we were to
+use an explicit (which we will not), we would have to investigate
+the eigenvalues of the spatial operator. If you followed the discussions
+of the video lectures
+(@dealiiVideoLectureSeeAlso{26,27,28})
+then you will remember that the pattern is that one needs to make sure
+that $k^s \propto h^t$ where $k$ is the time step, $h$ the mesh width,
+and $s,t$ are the orders of temporal and spatial derivatives.
+Whether you take the original equation ($s=1,t=2$) or the reformulation
+for only the real or imaginary part, the outcome is that we would need to
+choose $k \propto h^2$ if we were to use an explicit time stepping
+method. This is not feasible for the same reasons as in step-26 for
+the heat equation: It would yield impractically small time steps
+for even only modestly refined meshes. Rather, we have to use an
+implicit time stepping method and can then choose a more balanced
+$k \propto h$. Indeed, we will use the implicit Crank-Nicolson
+method as we have already done in step-25 before for the regular
+wave equation.
```

Comment:
Yes, I don't know either. I fixed the reference to step-23, but would like to avoid speculating about whether the C-N scheme conserves energy.

(I think that it might. The phase rotation part of the Strang splitting only, well, rotates the phase and so doesn't affect the energy. One would have to think about what happens with the the spatial half step. Separately: what should be preserved is actually the square of the wave function integrated over the domain. That's the particle number in the current case. It has the same form as the "energy" in other contexts.)